Directory indexing for creating search database

AasthaGupta opened the pull request:
**Fixes issue:**  #70
<!-- [Mention the issue number it fixes or add the details of the changes if it doesn't has a specific issue. -->


**Changes:**
<!-- Add here what changes were made in this awesome pull request. -->
This will ease the process of updating search entries because now we
don't have to maintain a hard-coded database and a simple update in the
directory will be enough.

<!-- Make sure that you enjoyed being the part of the OpenGenus Community. We would love to hear how can we make your contributing experience better. Thank you. Have a nice day! -->

AasthaGupta commented:
@AdiChat Please review it and let me know if any changes are required.

AdiChat commented:
Generating metadata `obj` every time Quark is loaded may not be efficient. 

Instead, you can create a script to generate the metadata (as you did) from a settings page `setting.html`. Users may trigger this manually. Also, moving the metadata from `popup.js` to a separate file will be a good design choice.

Kindly take a look into this and make appropriate changes. :+1:

AasthaGupta commented:
@AdiChat I have made the required changes. Please review it and let me know if any more changes are required.

AasthaGupta commented:
@Adichat any update on this?

AasthaGupta commented:
@AdiChat I have updated the pull request. 
- Now directory indexing will be done only once and stored in the local storage.Moreover, it will done in the background, so the extension can start without any delay.
- Added the option to `Refresh` under settings, to refresh the data.
